File: Ex-19/c++/Assunto.cpp
#include "Assunto.hpp"

Assunto::Assunto() : texto("")
{
}

Assunto::Assunto(const string &texto) : texto(texto)
{
}

Assunto::~Assunto()
{
}

void Assunto::setTexto(const string &texto)
{
  this->texto = texto;
}

const string &Assunto::getTexto() const
{
  return this->texto;
}

File: Ex-19/ex_19_java/Assunto.java
public class Assunto {
    private String texto = ""; // Inicializado para evitar NullPointerException

    /**
     * Define (sobrescreve) o texto completo do assunto.
     */
    public void setTexto(String texto) {
        this.texto = texto;
    }

    /**
     * Retorna o texto do assunto.
     */
    public String getTexto() {
        return this.texto;
    }
}